x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authz/interceptor/BulkShardRequestInterceptor.java

```diff
@@ -43,32 +43,33 @@ public BulkShardRequestInterceptor(ThreadPool threadPool, XPackLicenseState lice
     public void intercept(RequestInfo requestInfo, AuthorizationEngine authzEngine, AuthorizationInfo authorizationInfo,
                           ActionListener<Void> listener) {
         boolean shouldIntercept = licenseState.isSecurityEnabled();
-        var licenseChecker = new MemoizedSupplier<>(() -> licenseState.checkFeature(Feature.SECURITY_DLS_FLS));
+        var featureUsageChecker = new MemoizedSupplier<>(() -> licenseState.checkFeature(Feature.SECURITY_DLS_FLS));
         if (requestInfo.getRequest() instanceof BulkShardRequest && shouldIntercept) {
             IndicesAccessControl indicesAccessControl = threadContext.getTransient(AuthorizationServiceField.INDICES_PERMISSIONS_KEY);
-
-            final BulkShardRequest bulkShardRequest = (BulkShardRequest) requestInfo.getRequest();
-            for (BulkItemRequest bulkItemRequest : bulkShardRequest.items()) {
-                IndicesAccessControl.IndexAccessControl indexAccessControl =
-                    indicesAccessControl.getIndexPermissions(bulkItemRequest.index());
-                boolean found = false;
-                if (indexAccessControl != null) {
-                    boolean fls = indexAccessControl.getFieldPermissions().hasFieldLevelSecurity();
-                    boolean dls = indexAccessControl.getDocumentPermissions().hasDocumentLevelPermissions();
-                    if (fls || dls) {
-                        if (licenseChecker.get() && bulkItemRequest.request() instanceof UpdateRequest) {
+            BulkShardRequest bulkShardRequest = (BulkShardRequest) requestInfo.getRequest();
+            // this uses the {@code BulkShardRequest#index()} because the {@code bulkItemRequest#index()}
+            // can still be an unresolved date math expression
+            IndicesAccessControl.IndexAccessControl indexAccessControl = indicesAccessControl.getIndexPermissions(bulkShardRequest.index());
+            // TODO replace if condition with assertion
+            if (indexAccessControl != null) {
```

Comment:
This `if` must go away and be replaced with an assert, but the assert currently trips for enrich policies which make use of reindex.